fre/yamltools/combine_yamls.py

```diff
@@ -0,0 +1,265 @@
+
+def experiment_check(mainyaml_dir,comb,experiment):
+    """
+    Check that the experiment given is an experiment listed in the model yaml.
+    Extract experiment specific information and file paths.
+    Arguments:
+        mainyaml_dir  :  model yaml file
+        comb            :  combined yaml file name
+        experiment      :  experiment name
+    """
+    comb_model=yaml_load(comb)
+
+    # Check if exp name given is actually valid experiment listed in combined yaml
+    exp_list = []
+    for i in comb_model.get("experiments"):
+        exp_list.append(i.get("name"))
+
+    if experiment not in exp_list:
+        raise NameError(f"{experiment} is not in the list of experiments")
+
+    # set platform yaml filepath
+    if comb_model["shared"]["compile"]["platformYaml"] is not None:
+        py=comb_model["shared"]["compile"]["platformYaml"]
+        py_path=Path(os.path.join(mainyaml_dir,py))
+    else:
+        py=False
+
+    # Extract compile yaml path for exp. provided
+    # if experiment matches name in list of experiments in yaml, extract file path
+    for i in comb_model.get("experiments"):
+        if experiment == i.get("name"):
+            compileyaml=i.get("compile")
+            expyaml=i.get("pp")
+            analysisyaml=i.get("analysis")
+
+            if compileyaml is not None:
+                cy_path=Path(os.path.join(mainyaml_dir,compileyaml))
+            else:
+                cy_path=False
+            if expyaml is not None:
+                ey_path=[]
+                for e in expyaml:
+                    ey=Path(os.path.join(mainyaml_dir,e))
+                    ey_path.append(ey)
+            else:
+                ey_path=False
+            if analysisyaml is not None:
+                ay_path=[]
+                for a in analysisyaml:
+                    ay=Path(os.path.join(mainyaml_dir,a))
+                    ay_path.append(ay)
+            else:
+                ay_path=False
+
+            return (py_path,cy_path,ey_path,ay_path)
+
+def combine_compile(comb_m,compileyaml):
+    """
+    Combine compile yaml with the defined combined.yaml
+    Arguments:
+        comb_m       :  combined model yaml file
+        compileyaml  :  compile yaml file 
+    """
+    combined = comb_m
+
+    # copy compile yaml info into combined yaml
+    if compileyaml is not False:
+        with open(combined,'a',encoding='UTF-8') as f1:
+            with open(compileyaml,'r',encoding='UTF-8') as f2:
+                f1.write("\n### COMPILE INFO ###\n")
+                shutil.copyfileobj(f2,f1)
+        print(f"   compile yaml: {compileyaml}")
+
+def combine_platforms(comb_mc,platformsyaml):
+    """
+    Combine platforms yaml with the defined combined.yaml
+    Arguments:
+        comb_mc        : combined model and compile yaml file
+        platformsyaml  :  platforms yaml file
+    """
+    combined = comb_mc
+    # combine platform yaml
+    if platformsyaml is not False:
```

Comment:
see prev comment- is this pattern here for a GOOD reason? or is this a propagated issue?

if it's for a GOOD reason, that reason should be noted down somewhere nice and visible. otherwise we should change this to `XXXyaml is not None:` wherever that applies